fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneStoredFieldsTest.java

```diff
@@ -0,0 +1,395 @@
+    @BooleanSource
+    void testRollbackDeleteAllDocuments(boolean useOptimizedStoredFieldsFormat) throws Exception {
+        Index index = getSimpleTextIndex(useOptimizedStoredFieldsFormat);
+
+        try (FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, SIMPLE_DOC, index);
+            recordStore.saveRecord(createSimpleDocument(1623L, "Document 1", 2));
+            recordStore.saveRecord(createSimpleDocument(1624L, "Document 2", 2));
+            recordStore.saveRecord(createSimpleDocument(1547L, "NonDocument 3", 2));
+            context.commit();
+        }
+        try (FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, SIMPLE_DOC, index);
+            recordStore.deleteRecord(Tuple.from(1623L));
+            recordStore.deleteRecord(Tuple.from(1624L));
+            recordStore.deleteRecord(Tuple.from(1547L));
+            context.ensureActive().cancel();
+        }
+        try (FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, SIMPLE_DOC, index);
+            final RecordQuery query = buildQuery("Document", Collections.emptyList(), SIMPLE_DOC);
+            queryAndAssertFields(query, "text", Set.of(1623L, 1624L), Set.of("Document 1", "Document 2"));
+            if (useOptimizedStoredFieldsFormat) {
+                try (FDBDirectory directory = new FDBDirectory(recordStore.indexSubspace(index), context, index.getOptions())) {
+                    // When deleting all docs from the index, the last segment (_1) gets removed, leaving the _0 segment with the 3 tombstones
+                    assertDocCountPerSegment(directory, List.of("_0", "_1"), List.of(3, 0));
+                }
+                assertNull(timer.getCounter(LuceneEvents.Counts.LUCENE_DELETE_STORED_FIELDS));
+            }
+        }
+    }
+
+    @ParameterizedTest
+    @BooleanSource
+    void testComplexDocManyFields(boolean useOptimizedStoredFieldsFormat) throws Exception {
+        // Use a complex index with several fields
+        Index index = getComplexTextIndex(useOptimizedStoredFieldsFormat);
+
+        try (FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, COMPLEX_DOC, index);
+            recordStore.saveRecord(LuceneIndexTestUtils.createComplexDocument(1623L, "Hello", "Hello 2", 5, 12, false, 7.123));
+            recordStore.saveRecord(LuceneIndexTestUtils.createComplexDocument(1624L, "Hello record", "Hello record 2", 6, 13, false, 8.123));
+            recordStore.saveRecord(LuceneIndexTestUtils.createComplexDocument(1625L, "Hello record layer", "Hello record layer 2", 7, 14, true, 9.123));
+            context.commit();
+        }
+        try (FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, COMPLEX_DOC, index);
+            final RecordQuery query = buildQuery("record", Collections.emptyList(), COMPLEX_DOC);
+            // This doc has the OK as (Group, docID)
```

Comment:
What is this comment saying?

Reply:
Should be `PK` instead of `OK` - fixed.
This is just to let people know where the values in the assertions are coming from